CRM/Contact/BAO/Query.php

```diff
@@ -5829,12 +5811,40 @@ public function convertToPseudoNames(&$dao, $return = FALSE, $usedForAPI = FALSE
         $val = $dao->{$value['idCol']};
         if ($key == 'groups') {
           $dao->groups = $this->convertGroupIDStringToLabelString($dao, $val);
-          return;
+          continue;
```

Comment:
But, it's within a foreach loop - that seems to say 'outside a foreach loop'

    foreach ($this->_pseudoConstantsSelect as $key => $value) {